config/main.py

```diff
@@ -3153,111 +3214,155 @@ def del_vrf_vni_map(ctx, vrfname):
-        ctx.fail("argument is incomplete, nexthop not found!")
-    for i in range(0, len(command_str)):
-        if "nexthop" == command_str[i]:
-            prefix_str = command_str[:i]
-            nexthop_str = command_str[i:]
-    vrf_name = ""
-    cmd = 'sudo vtysh -c "configure terminal" -c "ip route'
-    if prefix_str:
-        if len(prefix_str) == 2:
-            prefix_mask = prefix_str[1]
-            cmd += ' {}'.format(prefix_mask)
-        elif len(prefix_str) == 4:
-            vrf_name = prefix_str[2]
-            prefix_mask = prefix_str[3]
-            cmd += ' {}'.format(prefix_mask)
+    config_db = ctx.obj['config_db']
+    key, route = cli_sroute_to_config(ctx, command_str)
+
+    # If defined intf name, check if it belongs to interface
+    if 'ifname' in route:
+        if (not route['ifname'] in config_db.get_keys('VLAN_INTERFACE') and
+            not route['ifname'] in config_db.get_keys('INTERFACE') and
+            not route['ifname'] in config_db.get_keys('PORTCHANNEL_INTERFACE')):
+            ctx.fail('interface {} doesn`t exist'.format(route['ifname']))
+
+    entry_counter = 1
+    if 'nexthop' in route:
+        entry_counter = len(route['nexthop'].split(','))
+
+    # Alignment in case the command contains several nexthop ip
+    for i in range(entry_counter):
+        if 'nexthop-vrf' in route:
+            if i > 0:
+                vrf = route['nexthop-vrf'].split(',')[0]
+                route['nexthop-vrf'] += ',' + vrf
         else:
-            ctx.fail("prefix is not in pattern!")
-    if nexthop_str:
-        if len(nexthop_str) == 2:
-            ip = nexthop_str[1]
-            if vrf_name == "":
-                cmd += ' {}'.format(ip)
-            else:
-                cmd += ' {} vrf {}'.format(ip, vrf_name)
-        elif len(nexthop_str) == 3:
-            dev_name = nexthop_str[2]
-            if vrf_name == "":
-                cmd += ' {}'.format(dev_name)
-            else:
-                cmd += ' {} vrf {}'.format(dev_name, vrf_name)
-        elif len(nexthop_str) == 4:
-            vrf_name_dst = nexthop_str[2]
-            ip = nexthop_str[3]
-            if vrf_name == "":
-                cmd += ' {} nexthop-vrf {}'.format(ip, vrf_name_dst)
-            else:
-                cmd += ' {} vrf {} nexthop-vrf {}'.format(ip, vrf_name, vrf_name_dst)
+            route['nexthop-vrf'] = ''
+
+        if not 'nexthop' in route:
+            route['nexthop'] = ''
+
+        if 'ifname' in route:
+            if i > 0:
+                route['ifname'] += ','
         else:
-            ctx.fail("nexthop is not in pattern!")
-    cmd += '"'
-    clicommon.run_command(cmd)
+            route['ifname'] = ''
+
+        # Set default values for distance and blackhole because the command doesn't have such an option
+        if 'distance' in route:
+            route['distance'] += ',0'
+        else:
+            route['distance'] = '0'
+
+        if 'blackhole' in route:
```

Comment:
We would need to handle one scenario for blackhole routes. If the user configure with "ifname" as "null", could you please add the attribute 'blackhole' as true.